src/requests.rs

```diff
@@ -0,0 +1,55 @@
+use crate::primitives::{
+    DecodableMessage, DecodeErrors, FromByte, HeaderRequest, Message, Request,
+};
+use bytes::Buf;
+use std::fmt::{Error, Formatter};
+
+#[derive(Debug)]
+// Kafka request to get the Api Version information of the Kafka broker
+pub struct RequestApiVersions {
+    // ToDo: move size and header out of the struct
+    // so that we can decode according to the header.api_key dynamically
+    // Maybe create struct for message body only (without size and header)?
+
+    // I'd like to include size in the requests because we may need information such as the
+    // average size of requests and show that on the dashboard
+    pub size: i32,
+    pub header: HeaderRequest,
+}
+
+impl Message for RequestApiVersions {
+    // Since self.size (an variable rather than a method of Message) is used, we cannot move the
+    // implementation of get_size() to the Message trait, and we have to copy this get_size
+    // implementation for all the Kafka requests, right?
```

Comment:
Yeah, that's correct, all Kafka messages which implement `Message` will have to implement this independently. Rust doesn't have inheritance.

As a suggestion, if there are multiple common fields that are shared between all messages, it might be worthwhile to put those in a struct that's a field on all the message structs. This is probably not worth it if `size` is the only such common field though.